### PR TITLE
Scalability blocks

### PR DESCRIPTION
**Pull request by mvcrouse**
## Related Issue
Supports #ISSUE_NUMBER

## Related PRs
This PR is not dependent on any other PR

## If applicable**
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by kpfadnis — CHANGES_REQUESTED** [review]
Left minor comments. Overall looks good.

File: fms_dgt/base/multiprocessing.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            List[ActorHandle]: List of workers for the block
+        """
+        return self._workers
+
+    def generate(self, *args, **kwargs):  # for interfacing with IL
+        return self(*args, **kwargs)
+
+    def __call__(self, inputs: DATASET_TYPE, *args: Any, **kwargs: Any) -> DATASET_TYPE:
+        # TODO: relax assumption that input is a list
+        partition_size = len(inputs) // len(self._workers)
```
[the remark this answers is not in the document]
Comment (mvcrouse) [reply fms_dgt/base/multiprocessing.py]: Yep good call, just fixed

File: fms_dgt/base/multiprocessing.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        # just return if empty input
+        if len(inputs) == 0:
+            return []
+
+        # TODO: relax assumption that input is a list
+        partition_size = max(len(inputs) // len(self._workers), 1)
+        actor_results = []
+        for i in range(len(self._workers)):
+            if i * partition_size >= len(inputs):
+                continue
```
Comment (kpfadnis) [inline fms_dgt/base/multiprocessing.py:74]: Isn't it better to just `break` out of the loop at this point where there are no more inputs to pass on to a worker?

File: fms_dgt/base/multiprocessing.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            DATASET_TYPE: Data after processing
+        """
+        # just return if empty input
+        if len(inputs) == 0:
+            return []
+
+        # TODO: relax assumption that input is a list
+        partition_size = max(len(inputs) // len(self._workers), 1)
+        actor_results = []
+        for i in range(len(self._workers)):
```
Comment (kpfadnis) [inline fms_dgt/base/multiprocessing.py:72]: For readability purposes, can we say

```python
for worker_idx in range(len(self._workers)):
```

**Review by kpfadnis — APPROVED** [review]
Looks good to me. 

@sivasankalpp had some reservations about worker management in `TENSOR PARALLEL` settings.

**Pull request merged**